include/bluetooth/services/latency.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ * @defgroup bt_gatt_latency BLE GATT Latency Service API
+ * @{
+ * @brief API for the BLE GATT Latency Service.
+ */
+
+#ifndef BT_GATT_LATENCY_H_
+#define BT_GATT_LATENCY_H_
+
+#include <bluetooth/uuid.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/gatt_dm.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Latency callback structure. */
+struct bt_gatt_latency_cb {
+	/** @brief Latency received callback.
+	 *
+	 * This function is called when a GATT write request has been received
+	 * by the Latency characteristic.
+	 *
+	 * @param[in] buf Latency data.
+	 */
+	void (*latency_request)(const void *buf, u16_t len);
+
+	/** @brief Latency received callback.
+	 *
+	 * This function is called when a GATT write response has been received
+	 * by the Latency characteristic.
+	 *
+	 * @param[in] buf Latency data.
+	 */
+	void (*latency_response)(const void *buf, u16_t len);
+};
+
+/** @brief Latency structure. */
+struct bt_gatt_latency {
+	/** Characteristic handle. */
+	u16_t handle;
+
+	/** Latency parameter. */
+	struct bt_gatt_write_params latency_params;
+
+	/** Connection object. */
+	struct bt_conn *conn;
+
+	/** Internal state. */
+	atomic_t state;
+};
+
+/** @brief Latency Service UUID. */
+#define LATENCY_UUID_DATA                                           \
+	((u8_t []) {0x3A, 0x81, 0xC0, 0x58, 0xDE, 0xFD, 0x46, 0x34, \
+		    0x9B, 0xF3, 0xDB, 0x58, 0x01, 0x6E, 0x13, 0x67})
+#define BT_UUID_LATENCY                                                     \
+	BT_UUID_DECLARE_128(0x3A, 0x81, 0xC0, 0x58, 0xDE, 0xFD, 0x46, 0x34, \
+			    0x9B, 0xF3, 0xDB, 0x58, 0x01, 0x6E, 0x13, 0x67)
+
+/** @brief UUID of the Latency Characteristic. **/
+#define BT_UUID_LATENCY_CHAR                                                \
+	BT_UUID_DECLARE_128(0x3A, 0x81, 0xC0, 0x58, 0xDE, 0xFD, 0x46, 0x34, \
+			    0x9B, 0xF3, 0xDB, 0x58, 0x02, 0x6E, 0x13, 0x67)
+
+/** @brief Initialize the GATT Latency Service.
+ *
+ *  @param[in] Latency service instance.
+ *  @param[in] cb Callbacks.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a negative error code is returned.
```

Comment:
Please use two `retval` here. One for `0`, one for `error`. (Mirror this edit elsewhere in this file.)